Dockerfile

```diff
@@ -2,60 +2,68 @@
-ENV FOOTPRINTS_DATA_DIR /opt/footprints_dir
-# Unset Java related ENVs since they may change with Oracle JDK
-ENV JAVA_VERSION=
-ENV JAVA_DEBIAN_VERSION=
-# Set JAVA_HOME to /usr/lib/jvm/default-java and link it to OpenJDK installation
-RUN ln -s /usr/lib/jvm/java-8-openjdk-amd64 /usr/lib/jvm/default-java
-ENV JAVA_HOME /usr/lib/jvm/default-java
+
+## Would you like to use Oracle JDK
 ARG ORACLE_JDK=false
+
+## Would you like to keep default Tomcat webapps
 ARG TOMCAT_EXTRAS=true
+
+## Would you like to install community modules
 ARG COMMUNITY_MODULES=true
 
+## Maximum Memory that Java can allocate
+ARG MAXIMUM_MEMORY="4G"
+
+## Initial Memory that Java can allocate
+ARG INITIAL_MEMORY="2G"
+
+RUN set -e \
+    export DEBIAN_FRONTEND=noninteractive \
+    dpkg-divert --local --rename --add /sbin/initctl \
+    apt-get -y update \
+    #Install extra fonts to use with sld font markers
+    apt-get install -y fonts-cantarell lmodern ttf-aenigma ttf-georgewilliams ttf-bitstream-vera ttf-sjfonts tv-fonts \
+        build-essential libapr1-dev libssl-dev default-jdk \
+    # Set JAVA_HOME to /usr/lib/jvm/default-java and link it to OpenJDK installation
+    && ln -s /usr/lib/jvm/java-8-openjdk-amd64 /usr/lib/jvm/default-java \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+ENV \
+    JAVA_HOME=/usr/lib/jvm/default-java \
+    DEBIAN_FRONTEND=noninteractive \
+    GEOSERVER_DATA_DIR=/opt/geoserver/data_dir \
+    GDAL_DATA=/usr/local/gdal_data \
+    LD_LIBRARY_PATH=/usr/local/gdal_native_libs:/usr/local/apr/lib:/opt/libjpeg-turbo/lib64 \
+    GEOSERVER_LOG_LOCATION=/opt/geoserver/data_dir/logs/geoserver.log \
+    FOOTPRINTS_DATA_DIR=/opt/footprints_dir \
+    GEOWEBCACHE_CACHE_DIR=/opt/geoserver/data_dir/gwc \
+    ENABLE_JSONP=true \
+    MAX_FILTER_RULES=20 \
+    OPTIMIZE_LINE_WIDTH=false \
+    GEOSERVER_OPTS="-Djava.awt.headless=true -server -Xms${INITIAL_MEMORY} -Xmx${MAXIMUM_MEMORY} -Xrs -XX:PerfDataSamplingInterval=500 \
+       -Dorg.geotools.referencing.forceXY=true -XX:SoftRefLRUPolicyMSPerMB=36000 -XX:+UseParallelGC -XX:NewRatio=2 \
+       -XX:+CMSClassUnloadingEnabled -Dfile.encoding=UTF8 -Duser.timezone=GMT -Djavax.servlet.request.encoding=UTF-8 \
+       -Djavax.servlet.response.encoding=UTF-8 -Duser.timezone=GMT -Dorg.geotools.shapefile.datetime=true" \
+       #-XX:+UseConcMarkSweepGC use this rather than parallel GC?
+    JAVA_OPTS="$JAVA_OPTS $GEOSERVER_OPTS" \
+    ## Unset Java related ENVs since they may change with Oracle JDK
+    JAVA_VERSION= \
+    JAVA_DEBIAN_VERSION= 
+
 
 WORKDIR /tmp/
+
+ADD logs $GEOSERVER_DATA_DIR
 ADD resources /tmp/resources
 ADD scripts /scripts
 RUN chmod +x /scripts/*.sh
-RUN /scripts/setup.sh
 ADD scripts/controlflow.properties $GEOSERVER_DATA_DIR
 ADD scripts/sqljdbc4-4.0.jar $CATALINA_HOME/webapps/geoserver/WEB-INF/lib/
 
-# Clean up APT when done.
-RUN echo "Yes, do as I say!" | apt-get remove --force-yes sed
-RUN echo "Yes, do as I say!" | apt-get remove --force-yes login
```

Comment:
Why did you remove this? Any reason

Reply:
Those needs to be executed earlier to have a real benefit, otherwise the previous layers will still contain the executables. I was not sure where `sed` or `login` were used in the previous steps

Reply:
I added back the `login` package removal in https://github.com/kartoza/docker-geoserver/pull/45/files#diff-3254677a7917c6c01f55212f86c57fbfR33, but not `sed` as it was commented in `develop`

Reply:
Thanks `sed` is now being used to substitute the new GeoServer admin password on runtime